src/basic/chattr-util.c

```diff
@@ -172,3 +173,56 @@ int read_attr_at(int dir_fd, const char *path, unsigned *ret) {
 
         return read_attr_fd(fd, ret);
 }
+
+int get_proj_id(int fd, uint32_t *ret) {
+        struct fsxattr attrs;
+
+        if (ioctl(fd, FS_IOC_FSGETXATTR, &attrs) < 0)
+                return -errno;
+
+        if (attrs.fsx_projid > 0)
+                *ret = attrs.fsx_projid;
+
```

Comment:
we have this rule that our functions must initialize all return parameters on succes (and leave the untouched on failure). here this means you should explicitly set `*ret = 0` or so if no projid is so far assigned.